python/python/lancedb/remote/table.py

```diff
@@ -481,15 +481,25 @@ def update(
         )
 
     def cleanup_old_versions(self, *_):
-        """cleanup_old_versions() is not supported on the LanceDB cloud"""
+        """
+        cleanup_old_versions() is not supported nor necessary on LanceDB Cloud.
+
+        Tables are automatically cleaned up and optimized.
+        """
         raise NotImplementedError(
-            "cleanup_old_versions() is not supported on the LanceDB cloud"
+            "cleanup_old_versions() is not supported on the LanceDB Cloud. "
```

Comment:
not a blocker since it's existing behavior, but should we raise an exception or just log a warning?
From a code migration perspective from local to cloud/enterprise, this means the user won't have to change their code. But if we don't raise an exception for Remote, would that present any footguns from a user perspective?